plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.test.ts

```diff
@@ -292,4 +335,118 @@ describe('fetch:template', () => {
       ).resolves.toEqual('["first","second","third"]');
     });
   });
+
+  describe('with extension=true', () => {
+    let context: ActionContext<FetchTemplateInput>;
+
+    beforeEach(async () => {
+      context = mockContext({
+        values: {
+          name: 'test-project',
+          count: 1234,
+          itemList: ['first', 'second', 'third'],
+        },
+        extension: true,
+      });
+
+      mockFetchContents.mockImplementation(({ outputPath }) => {
+        mockFs({
+          [outputPath]: {
+            'empty-dir-${{ values.count }}': {},
+            'static.txt': 'static content',
+            '${{ values.name }}.txt': 'static content',
```

Comment:
The `outputPath` here actually refers to the location where the action downloads the skeleton to, so these being unexpanded is right. The expanded paths are used in the `readFiles` call below.

Reply:
Hmm - I think we might be talking about different things 😅. All I'm suggesting is updating lines 356 and 357 to include a template variable that we _don't_ expect to be expanded in the file contents, and thereby confirming that the files that don't end in `.njk` are being copied without rendering.

Reply:
Ah, yes.

Reply:
So, I'm not testing that here, but I am testing that the `.njk` isn't processed when specifying `.jinja2` in https://github.com/backstage/backstage/pull/6800/files#diff-7282a1abb4b94b9336e1f996abdd3e2a5f179fc9ea631c868e1c0b453c57f918R423-R427.

Which is the code path you're looking for, right?

If so, not sure it's good to test that twice. I could just add a `.jinja2` here to make that symmetric?

Rambling, I'm going to, I like the idea to prevent silently hardcoding `jinja2` to always expand.